File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int size = s.size();
        int index = 0 , len = 0;

        function<int(int,int)> solve= [&](int low , int high){
            while(low >= 0 && high < size && s[low] == s[high]){
                low--;
                high++;
            }

            return high - low -1;
        };

        for(int i = 0 ; i < size; i++){
            int len1 = solve(i, i);
            int len2 = solve(i , i+1);

            int max_len  =max(len1 , len2);

            if(max_len  > len){
                len = max_len;
                index = i -(max_len -1)/2; 
            }
        }

        return s.substr(index, len);
    }
};

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution(object):

    def expand(self, s ,low, high):
        while low >= 0 and high < len(s) and s[low] == s[high]:
            low -= 1
            high +=1

        return high - low - 1 

    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        size = len(s)
        index = 0 
        length = 1

        for i  in range(0 , size):
            size1 = self.expand(s , i,i)
            size2 = self.expand(s , i,i+1)

            max_size = max(size1, size2)

            if max_size > length:
                length = max_size
                index = i - (length - 1) // 2 

        return s[index : index + length]


        